riscv-page-tables/src/page_table.rs

```diff
@@ -660,29 +659,38 @@ impl<T: PagingMode> PageTableInner<T> {
             .checked_add_pages_with_size(num_pages, page_size)
             .ok_or(Error::OutOfMapRange)?;
 
-        let iterator = start_addr
-            .iter_from_with_size(page_size)
-            .ok_or_else(|| Error::AddressMisaligned(start_addr.bits()))?;
+        let mut remaining = num_pages;
+        let mut vaddr = start_addr;
 
-        for vaddr in iterator.take(num_pages as usize) {
+        while remaining > 0 {
```

Comment:
Just for my own comprehension, you need this loop to handle non contiguous pages, is that correct?

Reply:
No, to handle indexes in the other table.